covigator/references/lineage_annotation.py

```diff
@@ -1,27 +1,468 @@
+        # Calculate codon for the last affected amino acid
+        if position_last:
+            codon_last = [position_last * 3 - 2, position_last * 3 - 1, position_last * 3]
+            last_aa = str(Seq(self.genome[(cds_start - 1) + codon_last[0] - 1: (cds_start - 1) + codon_last[2]]).translate())
+
+        # Create HGVS notations for different effects of the mutation
+        # HGVS for frameshift deletion
+        if deletion_length % 3 != 0:
+            hgvs = "p.{aa}{pos}fs".format(
+                aa=first_aa,
+                pos=protein_position
+            )
+            ref_aa = first_aa
+        # HGVS for a site where the deletion leads to an insertion of a new AA
+        elif delins:
+            hgvs = "p.{first_aa}{first_pos}_{last_aa}{last_pos}delins{ins}".format(
+                first_aa=first_aa,
+                first_pos=protein_position,
+                last_aa=last_aa,
+                last_pos=position_last,
+                ins=inserted_aa
+            )
+            # Return all AA that were deleted
+            ref_aa = str(Seq(self.genome[(cds_start - 1) + codon[0] - 1:
+                                         (cds_start - 1) + codon_last[2]]).translate())
+        # HGVS for deletion of one of multiple codons
+        else:
+            # Deletion spans multiple codons --> hgvs consists of first and last affected AA
+            if deletion_length / 3 > 1 and deletion_length % 3 == 0:
+                hgvs = "p.{first_aa}{first_pos}_{last_aa}{last_pos}del".format(
+                    first_aa=first_aa,
+                    first_pos=protein_position,
+                    last_aa=last_aa,
+                    last_pos=position_last
+                )
+                ref_aa = str(Seq(self.genome[(cds_start - 1) + codon[0] - 1:
+                                             (cds_start - 1) + codon_last[2]]).translate())
+            # Single codon deleted
+            else:
+                hgvs = "p.{aa}{pos}del".format(
+                    aa=first_aa,
+                    pos=protein_position
+                )
+                ref_aa = first_aa
+
+        return {"hgvs_p": hgvs,
+                "reference": ref_aa,
+                "alternate": "del",
+                "position": protein_position}
+
+    def get_hgvs_from_nuc_snp(self, variant) -> dict:
+        """
+        Translate pangolin nucleotide level SNPs description into hgvs protein annotation. Returns the variant
+        as hgvs string and the annotation of this variant
+
+        Args:
+            variant (dict): A dictionary with keys for genomic position, reference and alternate base
+
+        Returns:
+            annotation (dict): A dictionary with the annotation and HGVS notation of the SNP
+        """
+        position = variant.get("genomic_position")
+        ref_base = variant.get("reference")
+        alt_base = variant.get("alternate")
+        annotation = "synonymous"
```

Comment:
use the constant in `covigator/__init__.py`. Also, be careful with these terms, we use the controlled vocabulary from the Sequence Ontology (SO), eg: http://www.sequenceontology.org/browser/current_release/term/SO:0001583. The right term is "synonymous_variant". That was also the reason to introduce constants, to avoid this kind of entropy when typing the same string in different places. Please, check that others are too in agreement with SO